### PR TITLE
driver:platform:aducm:i2c add #include "error.h"

### PR DESCRIPTION
**Pull request by mchindri**
This is needed beacause the driver return SUCCESS and FAILURE
but the include is missing.

Signed-off-by: Mihail Chindris <mihail.chindris@analog.com>

### CODE REVIEW

File: drivers/platform/aducm3029/i2c.c [hunk 1]
```diff
@@ -42,6 +42,7 @@
 /******************************************************************************/
 
 #include "i2c.h"
+#include "error.h"
```
Comment (cristiantpop) [inline drivers/platform/aducm3029/i2c.c]: Why is this needed suddenly?

Comment (mchindri) [reply drivers/platform/aducm3029/i2c.c]: done

**Comment by mchindri** [comment]
V2
 - Update commit description
 - update pull request description
